### PR TITLE
MULE-19115: Optional parameter declared and annotated on SampleDataProvider is still defined as required in the model

### PR DESCRIPTION
**Pull request by gabrieldalborgo**

### CODE REVIEW

File: tests/test-extensions/sample-data-extension/src/main/java/org/mule/test/data/sample/extension/provider/GroupTestSampleDataProvider.java [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 import static org.mule.test.data.sample.extension.SampleDataExtension.adaptLegacy;
 
-import org.mule.runtime.extension.api.annotation.param.Optional;
+import org.mule.sdk.api.annotation.param.Optional;
```
Comment (marianogonzalez) [inline tests/test-extensions/sample-data-extension/src/main/java/org/mule/test/data/sample/extension/provider/GroupTestSampleDataProvider.java:11]: make sure to have tests that cover both cases... old and new `@Optional` annotations

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

File: tests/test-extensions/sample-data-extension/src/main/java/org/mule/test/data/sample/extension/provider/GroupTestSampleDataProvider.java (hunk 1 above)
Comment (gabrieldalborgo) [reply tests/test-extensions/sample-data-extension/src/main/java/org/mule/test/data/sample/extension/provider/GroupTestSampleDataProvider.java:11]: Yes, I had check it. There are both cases for VP's and SampleData

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/loader/java/type/ExtensionParameter.java [hunk 2]
```diff
@@ -79,23 +80,23 @@ default boolean shouldBeAdvertised() {
...
     final java.util.Optional<Optional> annotation = getAnnotation(Optional.class);
     if (annotation.isPresent()) {
-      final Optional optionalAnnotation = annotation.get();
-      final String defaultValue = optionalAnnotation.defaultValue();
-      if (!defaultValue.equals(Optional.NULL)) {
-        optionalDefaultValue = java.util.Optional.of(defaultValue);
-      }
+      return getDefaultValue(annotation.get());
+    }
+    final java.util.Optional<org.mule.sdk.api.annotation.param.Optional> sdkAnnotation =
```
Comment (ndinu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/loader/java/type/ExtensionParameter.java:94]: If both are present, do you think we should use the value from the extensions-api annotation?

Comment (gabrieldalborgo) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/loader/java/type/ExtensionParameter.java:94]: You're right. The other way makes more sense to me. I'll change it

**Review by ndinu — APPROVED** [review]

**Review by marianogonzalez — APPROVED** [review]

**Pull request merged**